2371/2/application.rb

```diff
@@ -0,0 +1,87 @@
+require 'optparse'
+require File.expand_path(File.dirname(__FILE__) + '/rap_battles_manager')
+require File.expand_path(File.dirname(__FILE__) + '/constants')
+
+# The Application responsible for start searching info
+class Application
+  attr_reader :options
+  def initialize
+    @thread = Thread.new do
```

Comment:
Ideally, in an initializer, you should set only the initial state.
All further logic will come after you send the appropriate message to your object.
I suggest you remove this loading functionality.